Documentation/README.md

```diff
@@ -57,3 +57,146 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
+```
+
+You can set `SOURCE_BUILD_SKIP_SUBMODULE_CHECK=1` environment variable or after you've pulled the changes, before you build, in your `source-build` repo just add the changes with, `git add src/corefx` and/or `git add src/coreclr`.
+
+Once the build is done, we will have a .NET Runtime containing the `Hashtable.PNSE()` API that can be used with our local cli in .NET Core projects.
+
+### How to use the produced Runtime
+
+The produced runtime will be under `bin/obj/{architecture}/{configuration}/blob-feed/packages`.
+
+Now create a new .NET Core project:
+```console
+dotnet new Console -o source-build-test
+```
+
+In your new project add a new NuGet.config to point to the Microsoft.NETCore.App and runtime packages.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <!-- NOTE: Leave this file here and keep it in sync with list in dir.props. -->
+  <!-- The command-line doesn't need it, but the IDE does.                    -->
+  <packageSources>
+    <clear/>
+    <add key="local feed" value="C:\repos\source-build\bin\obj\x64\Release\blob-feed\packages" />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+  </packageSources>
+</configuration>
+```
+
+Then in your csproj we need to set the Runtime version:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
+    <RootNamespace>source_build_test</RootNamespace>
+    <RuntimeFrameworkVersion>2.2.0-preview1-26509-04</RuntimeFrameworkVersion>
+    <NETCoreAppMaximumVersion>2.2</NETCoreAppMaximumVersion>
+  </PropertyGroup>
+
+</Project>
+```
+
+Note that the `TargetFramework` value needs to be the minimum or later framework that the Runtime supports. Therefore if there is no cli produced yet or you don't have installed a dotnet cli that supports that `TargetFramework` you need to add the property `NETCoreAppMaximumVersion` to match the version framework you want to target.
+
+If you don't set this property, you will see an error similar to this one:
+```console
+C:\Program Files\dotnet\sdk\2.1.300\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(137,5): error : The current .NET SDK does not support targeting .NET Core 2.2.  Either target .NET Core 2.1 or lower, or use a version of the .NET SDK that supports .NET Core 2.2.
+```
+
+Also, `RuntimeFrameworkVersion` needs to match whatever version of the runtime was produced.
+
+After that you're all set to use your new API and test it with a locally produced runtime.
+
+```cs
+using System;
+using System.Collections;
+
+namespace source_build_test
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Hashtable.PNSE();
+        }
+    }
+}
+```
+
+In order to run the app you have two options:
+
+1. Publish the app:
+
+  - Run `dotnet publish`.
+  - Run `.exe` in publish directory.
```

Comment:
This shows up pretty badly, suggestion:

```markdown
1. Publish the app:
   - Run `dotnet publish`.
   - Run `.exe` in publish directory.
2. Use `dotnet run`:
   - In order to use `dotnet run` you will have to install the produced runtime before running the app. Note that installing the runtime, whenever a real runtime for the current produced version is released, if you forget to delete/uninstall the custom runtime, it could cause issues.
```

(`-` needs to be aligned with the beginning of the `Publish` word after `1.` to be inside the numbered list.)